imap_processing/spice/geometry.py

```diff
@@ -87,3 +90,148 @@ def imap_state(
         SpiceBody.IMAP.name, et, ref_frame.name, "NONE", observer.name
     )
     return np.asarray(state)
+
+
+def generate_spin_data(start_met: float, end_met: Optional[float] = None) -> dict:
+    """
+    Generate a spin table CSV covering one or more days.
+
+    Spin table contains the following fields:
+        (
+        spin_number,
+        spin_start_sec,
+        spin_start_subsec,
+        spin_period_sec,
+        spin_period_valid,
+        spin_phas_valid,
+        spin_period_source,
+        thruster_firing
+        )
+    This function creates spin data using start MET and end MET time.
+    Each spin start data uses the nominal 15 second spin period. The spins that
+    occur from 00:00(Mid-night) to 00:10 UTC are marked with flags for
+    thruster firing, invalid spin period, and invalid spin phase.
+
+    Parameters
+    ----------
+    start_met : float
+        Provides the start time in Mission Elapsed Time (MET).
+    end_met : float
+        Provides the end time in MET. If not provided, default to one day
+        from start time.
+
+    Returns
+    -------
+    spin_df : dict
+        Spin data. May need to save data to CSV file.
+    """
+    if end_met is None:
+        # end_time is one day after start_time
+        end_met = start_met + 86400
+
+    # Create spin start second data of 15 seconds increment
+    spin_start_sec = np.arange(start_met, end_met, 15)
+
+    spin_dict = {
+        "spin_number": np.arange(spin_start_sec.size, dtype=np.uint32),
+        "spin_start_sec": spin_start_sec,
+        "spin_start_subsec": np.full(spin_start_sec.size, 0, dtype=np.uint32),
+        "spin_period_sec": np.full(spin_start_sec.size, 15.0, dtype=np.float32),
+        "spin_period_valid": np.ones(spin_start_sec.size, dtype=np.uint8),
+        "spin_phas_valid": np.ones(spin_start_sec.size, dtype=np.uint8),
+        "spin_period_source": np.zeros(spin_start_sec.size, dtype=np.uint8),
+        "thruster_firing": np.zeros(spin_start_sec.size, dtype=np.uint8),
+    }
+
+    # Convert spin_start_sec to datetime to set repointing times flags
+    spin_start_dates = met_to_j2000ns(spin_start_sec)
+    spin_start_dates = cdflib.cdfepoch.to_datetime(spin_start_dates)
+
+    # Convert DatetimeIndex to Series for using .dt accessor
+    spin_start_dates_series = pd.Series(spin_start_dates)
+
+    # Find index of all timestamps that fall within 10 minutes after midnight
+    repointing_times = spin_start_dates_series[
```

Comment:
If we plan on using this test table w/ SPICE (and therefore use the test ck kernel IMAP_spacecraft_attitude.py) then here you might account for the test ck kernel's gaps that occur around midnight each day. I have been using the start/stop of the intervals within the kernel as the repointing times. I think you might eventually have to do that here. @subagonsouth what do you think?

Reply:
I am envisioning that in-situ will use this function but good point. I will need to think about how to alter `get_spacecraft_spin_phase` to use what you guys have vs this. I can think about this offline and that work will happen in upcoming PR.